### PR TITLE
Result as status

### PR DESCRIPTION
**Pull request by OCHyams**
NOTE: This is based on PR #22.

This change introduces the --fail-lt <float> argument for the test tool The
program will exit with FAIL(2) if the test result is less than the specified
value.

The default value for the argument is 0.0; All tests will terminate with the
exit code OK(0).

### CODE REVIEW

**Comment by OCHyams** [comment]
@jmorse what do you think of this?

If the use case is always going to be `--fail-lt 1.0` it might make more sense to have an arg which doesn't take a threshold value.

The flexibility offered here may just be causing unnecessary clutter?

File: dex/tools/no_tool_/Tool.py [hunk 1]
```diff
@@ -58,5 +59,7 @@ def handle_options(self, defaults):
             raise Error('<d>no subtool specified</>\n\n{}\n'.format(
                 self.parser.format_help()))
 
-    def go(self):
-        return 1
+    def go(self) -> ReturnCode:
+        # This fn is never called because not specifying a subtool raises an
+        # exception.
+        return ReturnCode.OK
```
Comment (jmorse) [inline dex/tools/no_tool_/Tool.py]: Wouldn't it make sense to return an error in that case though?

Comment (OCHyams) [reply dex/tools/no_tool_/Tool.py]: I've fixed this in #22.

**Comment by OCHyams** [comment]
I'll rebase this once #22 has been accepted.

**Comment by OCHyams** [comment]
Rebased.

**Pull request merged**